admin/class-aioseop-helper.php

```diff
@@ -0,0 +1,1128 @@
+<?php
+/**
+ * AIOSEOP Helper/Info Class
+ *
+ * Description. (use period)
+ *
+ * @link URL
+ *
+ * @package All-in-One-SEO-Pack
+ * @since 2.4.2
+ */
+
+/**
+ * All in One SEO Plugin Helper
+ *
+ * @since 2.4.2
+ */
+class AIOSEOP_Helper {
+	/**
+	 * Help Text for jQuery UI Tooltips.
+	 *
+	 * @since 2.4.2
+	 * @var array $help_text {
+	 *     @type string
+	 * }
+	 */
+	private $help_text = array();
+
+	/**
+	 * Constructor
+	 *
+	 * @since 2.4.2
+	 *
+	 * @param string $module Module/Class name.
+	 */
+	public function __construct( $module = '' ) {
+		if ( current_user_can( 'aiosp_manage_seo' ) ) {
+			$this->_set_help_text( $module );
+		}
+	}
+
+	/**
+	 * Set this Help Text
+	 *
+	 * Sets the Help Text according to the module/class in use, but if there is
+	 * no class name in $module, then this Help Text will add all module help texts.
+	 *
+	 * @ignore
+	 * @since 2.4.2
+	 * @access private
+	 *
+	 * @param string $module All_in_One_SEO_Pack module.
+	 */
+	private function _set_help_text( $module ) {
+
+		switch ( $module ) {
+			case 'All_in_One_SEO_Pack':
+				$this->help_text = $this->help_text_general();
+				$this->help_text = array_merge( $this->help_text, $this->help_text_post_meta() );
+				break;
+			case 'All_in_One_SEO_Pack_Performance':
+				$this->help_text = $this->help_text_performance();
+				break;
+			case 'All_in_One_SEO_Pack_Sitemap':
+				$this->help_text = $this->help_text_sitemap();
+				break;
+			case 'All_in_One_SEO_Pack_Opengraph':
+				$this->help_text = $this->help_text_opengraph();
+				break;
+			case 'All_in_One_SEO_Pack_Robots':
+				$this->help_text = $this->help_text_robots_generator();
+				break;
+			case 'All_in_One_SEO_Pack_File_Editor':
+				$this->help_text = $this->help_text_file_editor();
+				break;
+			case 'All_in_One_SEO_Pack_Importer_Exporter':
+				$this->help_text = $this->help_text_importer_exporter();
+				break;
+			case 'All_in_One_SEO_Pack_Bad_Robots':
+				$this->help_text = $this->help_text_bad_robots();
+				break;
+			default:
+				$this->help_text = array_merge( $this->help_text, $this->help_text_general() );
+				$this->help_text = array_merge( $this->help_text, $this->help_text_performance() );
+				$this->help_text = array_merge( $this->help_text, $this->help_text_sitemap() );
+				$this->help_text = array_merge( $this->help_text, $this->help_text_opengraph() );
+				$this->help_text = array_merge( $this->help_text, $this->help_text_robots_generator() );
+				$this->help_text = array_merge( $this->help_text, $this->help_text_file_editor() );
+				$this->help_text = array_merge( $this->help_text, $this->help_text_importer_exporter() );
+				$this->help_text = array_merge( $this->help_text, $this->help_text_bad_robots() );
+				$this->help_text = array_merge( $this->help_text, $this->help_text_post_meta() );
```

Comment:
isn't this too wordy? array_merge can accept multiple parameters and might have better performance

Reply:
Changed